### PR TITLE
Add API Docs Contribution page

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: Develop/Collaborate/API_Docs_Contributions/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+title: API Docs Contribution
+permalink: Develop/Collaborate/API_Docs_Contribution/
+parent: Collaborate
+layout: default
+nav_order: 500
+---
+
+The preferred way of writing API documentation for Qt-based APIs is with the help of the [QDoc](https://doc.qt.io/qt-5/qdoc-index.html) documentation tool. Its use for documenting Sailfish APIs is supported with the [sailfish-qdoc-template](https://sailfishos.org/develop/docs/sailfish-components-pickers/) package, which also comes with essential usage notes. What follows here is a collection of advanced tips, recommendations and API docs writing conventions.
```
Comment (rainemak) [inline Develop/Collaborate/API_Docs_Contributions/README.md:9]: The sailfish-qdoc-template points to https://sailfishos.org/develop/docs/sailfish-components-pickers/ which is a wrong link

"Its use for documenting Sailfish APIs is supported with the [sailfish-qdoc-template](https://sailfishos.org/develop/docs/sailfish-components-pickers/) package, which also comes with essential usage notes."

I'd rephrase above sentence (with right links of course). 
```
The [QDoc](https://doc.qt.io/qt-5/qdoc-index.html) usage on Sailfish is supported with the [Sailfish QDoc template](https://sailfishos.org/develop/docs/sailfish-components-pickers/), which comes with essential usage notes.
```

File: Develop/Collaborate/API_Docs_Contributions/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+### Various tips on writing QDoc based documentation
```
Comment (rainemak) [inline Develop/Collaborate/API_Docs_Contributions/README.md:11]:
```suggestion
### Tips on writing QDoc based documentation
```
Better without extra various.

**Review by rainemak — CHANGES_REQUESTED** [review]
Very nice and informative piece for documenting APIs. Added two small remarks, let's discuss those through, fix them and land this.

File: Develop/Collaborate/API_Docs_Contributions/README.md (hunk 1 above)
Comment (martyone) [reply Develop/Collaborate/API_Docs_Contributions/README.md]: Updated. Did you really mean just "on Sailfish"? Without "OS" or something after "Sailfish".

Comment (rainemak) [reply Develop/Collaborate/API_Docs_Contributions/README.md]: "Sailfish OS" works better.

File: Develop/Collaborate/API_Docs_Contributions/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+---
+title: API Docs Contribution
+permalink: Develop/Collaborate/API_Docs_Contribution/
+parent: Collaborate
+layout: default
+nav_order: 500
+---
+
+The preferred way of writing API documentation for Qt-based APIs is with the help of the [QDoc](https://doc.qt.io/qt-5/qdoc-index.html) documentation tool. [QDoc](https://doc.qt.io/qt-5/qdoc-index.html) usage on Sailfish is supported with the [Sailfish QDoc template](https://sailfishos.org/develop/docs/sailfish-qdoc-template/), which comes with essential usage notes. What follows here is a collection of advanced tips, recommendations and API docs writing conventions.
```
Comment (rainemak) [inline Develop/Collaborate/API_Docs_Contributions/README.md:9]: Is https://sailfishos.org/develop/docs/sailfish-qdoc-template/ on it way as well? Currently it's 404 not found.

Comment (martyone) [reply Develop/Collaborate/API_Docs_Contributions/README.md]: yes, jb#58373

**Review by rainemak — APPROVED** [review]

File: Develop/Collaborate/API_Docs_Contributions/README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+The [url](https://doc.qt.io/qt-5/25-qdoc-configuration-derivedprojects.html#url) variable is meant to point to the location where the documentation is going to be published. It is used as a base to form external links to the documentation module when it is refered by other modules.
+
+Bad example:
+
+```
+url = https://github.com/sailfishos/foo-bar
+```
+
+Less bad example would use `https://sailfishos.org/develop/docs/foo-bar` as the URL. This would be already correct location at the time of writing this documentation, but you should also utilize the `BASE_URL` variable to enable easy relocation of the online documentation.
```
Comment (vigejolla) [inline Develop/Collaborate/API_Docs_Contributions/README.md:110]: Where is this `BASE_URL` defined?

Comment (martyone) [reply Develop/Collaborate/API_Docs_Contributions/README.md:110]: https://github.com/sailfishos/sailfish-qdoc-template/blob/master/config/sailfish-qdoc-template.prf#L22

**Comment by martyone** [comment]
Rebased

**Comment by martyone** [comment]
To be merged after next release with sailfish-qdoc-template updated.

File: Develop/Collaborate/API_Docs_Contributions/README.md (hunk 4 above)
Comment (vigejolla) [reply Develop/Collaborate/API_Docs_Contributions/README.md:110]: > https://github.com/sailfishos/sailfish-qdoc-template/blob/master/config/sailfish-qdoc-template.prf#L22

I thought it would be defined in the template, but for some reason I didn't find it. Maybe I was tired or something... nevermind :)

**Comment by vigejolla** [comment]
rebased

**Review by rainemak — APPROVED** [review]
